File: tiny-event-sourcing-spring-app/src/main/kotlin/ru/quipy/shop/order/events/OrderPaymentSetEvent.java
package ru.quipy.shop.order.events;


import lombok.Getter;
import ru.quipy.core.annotations.DomainEvent;
import ru.quipy.domain.Event;
import ru.quipy.shop.order.OrderAggregate;

import java.util.UUID;

@DomainEvent(name = "ORDER_PAYMENT_LINKED")
public class OrderPaymentSetEvent extends Event<OrderAggregate> {

    @Getter
    private final UUID orderId;

    @Getter
    private final UUID paymentId;

    public OrderPaymentSetEvent(UUID orderId, UUID paymentId) {
        super();
        this.orderId = orderId;
        this.paymentId = paymentId;
    }
}

File: tiny-event-sourcing-spring-app/src/main/kotlin/ru/quipy/shop/order/events/OrderPaymentSetEvent.kt
package ru.quipy.shop.order.events


import lombok.Getter
import ru.quipy.core.annotations.DomainEvent
import ru.quipy.domain.Event
import ru.quipy.shop.order.OrderAggregate

import java.util.UUID

@DomainEvent(name = "ORDER_PAYMENT_LINKED")
class OrderPaymentSetEvent(
    @Getter
    public val orderId: UUID,
    @Getter
    public val paymentId: UUID
) : Event<OrderAggregate>(
    name = "ORDER_PAYMENT_LINKED"
)